### ISSUE TITLE
suggestion xfxEditor

### ISSUE THREAD

**Issue by 14skywalker** [XamFormsExtended/Xfx.Controls#20]
Title says it all: we would welcome a xfxEditor

**Comment by ChaseFlorell** [comment]
@14skywalker I like the idea. Are you simply looking for an Editor with floated labels and an error message, or do you envision more?

**Comment by 14skywalker** [comment]
Hi @ChaseFlorell, interesting that you ask. We were also looking for an editor that would increase in height when needed. Like this: https://storage.googleapis.com/material-design/publish/material_v_11/assets/0B5ZSepuCX1xOVEhXVDltQzgzQzA/TextFields-MultiLine-Long-01.mp4

**Comment by ChaseFlorell** [comment]
@14skywalker you can actually do that now by first [measuring the length of the text](https://gist.github.com/rudyryk/7b9633e847e727d12de7) and adjusting the `HeightRequest` when the height changes. For added pizazz, I would use an [`Animation`](https://developer.xamarin.com/guides/xamarin-forms/user-interface/animation/custom/) to adjust the height.

**Comment by 14skywalker** [comment]
Thanks @ChaseFlorell, This would mean that the XfxEditor would only add the possibility to use new lines in the input? That is something we can live without.

**Comment by 14skywalker** [comment]
I now see that with "you can do that now" you mean with the Editor (not currently in combination with a floating label). Meaning: We still would welcome an XfxEditor.

**Comment by ChaseFlorell** [comment]
Yes that's correct. We can do the editor sizing now using a behavior. The floated labels would need implementing.

**Comment by 14skywalker** [comment]
I did make a start with an XfxEditor and Android renderer, but I could not make the `EditorEditText` inside the `TextInputLayout` to use more space than one line. By the way, the `EditorEditText` is more protected than the `EditText` that is used by the XfxEntry on android. I needed reflection to get to the constructor. Maybe there is a better way... So, for what it is worth:

[XfxEditor.cs](https://github.com/XamFormsExtended/Xfx.Controls/files/1124159/XfxEditor.cs.txt)
[XfxEditorRenderer.cs](https://github.com/XamFormsExtended/Xfx.Controls/files/1124165/XfxEditorRenderer.cs.txt)
